### PR TITLE
Handle ConferenceIq in the IO pool (and cleanup).

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: src/test/java/org/jitsi/jicofo/OSGiHandler.java [hunk 1]
```diff
@@ -72,6 +73,12 @@ public MockXmppProvider getXmppProvider()
 
     public void shutdown()
     {
+        System.clearProperty("org.jitsi.jicofo.PING_INTERVAL");
+        System.clearProperty(XmppClientConnectionConfig.legacyXmppDomainPropertyName);
+        System.clearProperty(XmppClientConnectionConfig.legacyDomainPropertyName);
+        System.clearProperty(XmppClientConnectionConfig.legacyUsernamePropertyName);
+        JitsiConfig.Companion.reloadNewConfig();
+
```
Comment (JonathanLennox) [inline src/test/java/org/jitsi/jicofo/OSGiHandler.java:81]: This doesn't logically seem part of the "Move XmppTest" commit - what's it doing?

File: src/main/kotlin/org/jitsi/jicofo/JicofoServices.kt [hunk 2]
```diff
@@ -201,7 +184,7 @@ open class JicofoServices {
             it.stop()
         }
         healthChecker?.stop()
-        channelAllocationExecutor.shutdownNow()
+        TaskPools.ioPool.shutdownNow()
```
Comment (JonathanLennox) [inline src/main/kotlin/org/jitsi/jicofo/JicofoServices.kt:179]: I feel like it'd be cleaner to have a single shutdown method in the `TaskPools` companion object.

File: spotbugs-exclude.xml [hunk 3]
```diff
@@ -26,6 +26,7 @@
     <Or>
       <Bug pattern="RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT"/>
       <Bug pattern="NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE"/>
+      <Bug pattern="DE_MIGHT_IGNORE"/>
```
Comment (JonathanLennox) [inline spotbugs-exclude.xml:29]: What warning are you suppressing here?

File: src/test/java/org/jitsi/jicofo/OSGiHandler.java (hunk 1 above)
Comment (bgrozev) [reply src/test/java/org/jitsi/jicofo/OSGiHandler.java:81]: Moving the test changed the order in which tests are run, which exposed a bug with some other tests changing the configuration. Now that I look at it more carefully (though I still don't understand it fully), it should probably be JicofoConfigTest that needs to clean-up after itself.

File: spotbugs-exclude.xml (hunk 3 above)
Comment (bgrozev) [reply spotbugs-exclude.xml:29]: It doesn't need to be suppressed anymore, removing.

File: src/test/java/org/jitsi/jicofo/OSGiHandler.java (hunk 1 above)
Comment (bgrozev) [reply src/test/java/org/jitsi/jicofo/OSGiHandler.java:81]: Aha. It used to be that `OSGiHandler.init` never ran before `XmppConfigTest`. But now `AllocateConferenceTest` runs prior to `XmppConfigTest`, and if `OSGiHandler` leaves these properties behind it brakes things.

**Review by JonathanLennox — APPROVED** [review]

**Pull request merged**